assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useReducer } from "react"
+import { loadState, saveState } from "../localStorage"
+import { Dispatch, Reducer, State } from "../state"
+
+const APP_STATE_KEY = "mbta-skate-state"
+
+const PERSISTED_KEYS = ["selectedRouteIds"]
+
+const usePersistedStateReducer = (
+  reducer: Reducer,
+  defaultValue: State
+): [State, Dispatch] => {
+  const [state, dispatch] = useReducer(
+    reducer,
+    defaultValue,
+    (initial: State) => ({ ...initial, ...loadState(APP_STATE_KEY) })
+  )
+
+  useEffect(() => {
+    saveState(APP_STATE_KEY, state, PERSISTED_KEYS)
+  }, [state])
```

Comment:
If you moved the `filter` to this file, then the filtered object could also be used here and we'd only call `saveState` when the saved part of the state changes.

Reply:
To prevent calling `saveState` on changes we don't care about, this line should also reference `persistableState` instead of `state` (and `persistableState` will need to be calculated before `useEffect`)

Also, typo `perstableState` -> `persistableState`.